pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/BlockExchange.java

```diff
@@ -237,14 +249,10 @@ private void sendPrivate(MseBlock block, List<DataBuffer> serializedStats)
         _earlyTerminated = BlockExchange.this.send(((MseBlock.Data) block));
       } else {
         _earlyTerminated = BlockExchange.this.send(((MseBlock.Eos) block), serializedStats);
+        _completed = true;
```

Comment:
Should we set it in `public void send(MseBlock.Eos block, List<DataBuffer> serializedStats)`?